core/src/main/java/jenkins/model/ParameterizedJobMixIn.java

```diff
@@ -218,7 +218,16 @@ public final void doBuild(StaplerRequest req, StaplerResponse rsp, @QueryParamet
 
         Queue.Item item = Jenkins.get().getQueue().schedule2(asJob(), delay.getTimeInSeconds(), getBuildCause(asJob(), req)).getItem();
         if (item != null) {
-            rsp.sendRedirect(SC_CREATED, req.getContextPath() + '/' + item.getUrl());
+            if (req.getHeader("User-Agent") != null && req.getHeader("User-Agent").contains("Mozilla/5.0")) {  // Using a web browser
```

Comment:
I don't see a need to check the user agent, shouldn't we be able to always redirect?

Reply:
Would it make sense if I send a POST to /jobName/build by curl and then get the job's page content as response?

Reply:
i think it would be fine although would be better if we could do the build page but i don't think that's too practical here

Reply:
I found another issue about your suggestion. If the page is responded, the status code is 200 instead of 201. Is it okay?

Reply:
is it safe to redirect to the builds page instead? that would make more sense posting to build redirecting you to the page the build will be on.

I tested it with pipeline and it seemed fine but i'm not sure if there's some cases where it wouldn't work

Reply:
I'm afraid that programmers designed a program to POST to /jobName/build, and check if the status code is 201 to determine if the POST action is successful. If this POST now always redirect to the job page with the 200 status code, those programs might fail.

Therefore, what I do is to check if the `/jobName/build` is visited by browser or not. If visited by browser, then redirect to the job page with 200; if not, then response with 201.

Reply:
any other thoughts @jenkinsci/core-pr-reviewers ?

Reply:
> redirecting to the same page: the build page

Build now creates (or reuses) a queue item, not a build. There are many reasons no build may exist (immediately) after you click _Build Now_.